internal/gr.go

```diff
@@ -135,13 +139,13 @@ func (g *GRGetter) GetWork(ctx context.Context, workID int64) (_ []byte, authorI
 
 	Log(ctx).Debug("getting book", "bookID", bookID)
 
-	out, _, authorID, err := g.GetBook(ctx, bookID)
+	out, _, authorID, err := g.GetBook(ctx, bookID, loadEditions)
 	return out, authorID, err
 }
 
 // GetBook fetches a book (edition) from GR.
-func (g *GRGetter) GetBook(ctx context.Context, bookID int64) (_ []byte, workID, authorID int64, _ error) {
-	if workBytes, ttl, ok := g.cache.GetWithTTL(ctx, BookKey(bookID)); ok && ttl > 0 {
+func (g *GRGetter) GetBook(ctx context.Context, bookID int64, loadEditions editionsCallback) (_ []byte, workID, authorID int64, _ error) {
+	if workBytes, ttl, ok := g.cache.GetWithTTL(ctx, BookKey(bookID)); ok && ttl > 0 && loadEditions == nil {
```

Comment:
We don't currently refresh works more often than books because `GetWork` is just `GetBook(bestBookID)`. This will cause us to start refreshing those best books more often, which I think is good.